### PR TITLE
MITM communication

### PR DESCRIPTION
**Pull request by Fabio1988**
provide better error events on account fetching.

RDM should not return http 404 when account is not found, it should return 200 with proper error codes... needs testing with MITM.... I am in contact with devs.

e.g.
```json
{
  "status": "error",
  "error": "accountNotFound"
}
```
Possible error events:
- accountNotFound
- noAccountLeft
- deviceNotFound
- instanceNotFound
 - noTaskLeft

### CODE REVIEW

**Comment by 123FLO321** [comment]
Does the api still respond with the correct http status code?

**Comment by Fabio1988** [comment]
> Does the api still respond with the correct http status code?

No, because none of the MITM can provide handling for 404.... It's a tricky workaround for them and mostly not handled.. some MITM even crash/restart on a 404.. I will write you a DM

**Comment by Fabio1988** [comment]
Or should we still use status codes like NOT_FOUND 404 with proper error in response?

File: Sources/RealDeviceMapLib/Webhook/WebHookRequestHandler.swift [hunk 1]
```diff
@@ -1022,6 +1022,9 @@ public class WebHookRequestHandler {
                         mysql: mysql, uuid: uuid, username: username,
                         account: account, timestamp: timestamp
                     )
+                    if task.isEmpty {
+                        response.respondWithError(event: .taskNotFound)
```
Comment (123FLO321) [inline Sources/RealDeviceMapLib/Webhook/WebHookRequestHandler.swift:1026]: rename to `noTaskLeft`

File: Sources/RealDeviceMapLib/Webhook/WebHookRequestHandler.swift [hunk 2]
```diff
@@ -1057,7 +1060,7 @@ public class WebHookRequestHandler {
                     guard let newAccount = try InstanceController.global.getAccount(mysql: mysql, deviceUUID: uuid)
                     else {
                         Log.error(message: "[WebHookRequestHandler] [\(uuid)] Failed to get account for \(uuid)")
-                        response.respondWithError(status: .notFound)
+                        response.respondWithError(event: .accountNotFound)
```
Comment (123FLO321) [inline Sources/RealDeviceMapLib/Webhook/WebHookRequestHandler.swift:1063]: rename to `noAccountLeft`

**Review by 123FLO321 — CHANGES_REQUESTED** [review]

File: Sources/RealDeviceMapLib/Webhook/WebHookRequestHandler.swift (hunk 2 above)
Comment (123FLO321) [reply Sources/RealDeviceMapLib/Webhook/WebHookRequestHandler.swift:1063]: (only this occurrence. other occurrences can stay as `accountNotFound`)

**Comment by Fabio1988** [comment]
any more thoughts on that?

**Comment by Fabio1988** [comment]
Mentioned thousand times, time to merge

**Pull request merged**